tdrs-backend/tdpservice/reports/views.py

```diff
@@ -0,0 +1,58 @@
+"""Check if user is authorized."""
+import logging
+
+from rest_framework import mixins, viewsets
+from rest_framework.permissions import AllowAny
+
+from ..users.permissions import IsUserOrReadOnly
+from .models import User
+from .serializers import ReportFileSerializer
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+# class AuthorizationCheck(APIView):
+#     """Check if user is authorized."""
+
+#     query_string = True
+#     pattern_name = "authorization-check"
+#     permission_classes = [AllowAny] # What should I set this to?
+
+#     def post(self, request, *args, **kwargs):
+#         """Handle get request and verify user is authorized."""
+#         user = request.user
+#         serializer = ReportFileSerializer(request.data)
+#         if user.is_authenticated:
+#             logger.info(
+#                 "Auth check PASS for user: %s on %s", user.username, timezone.now()
+#             )
+#             res = Response({"success":True})
+#             res['Access-Control-Allow-Headers'] = "X-CSRFToken"
+#             return res
+#         else:
+#             logger.info("Auth check FAIL for user on %s", timezone.now())
+#             return Response({"authenticated": False})
+
+
+class ReportFileViewSet(
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet,
+):
+    """Report file views."""
+
+    queryset = User.objects.select_related("stt")
+
+    def get_permissions(self):
+        """Get permissions for the viewset."""
+        if self.action == "create":
```

Comment:
@riatzukiza I believe this should be limited to Data Preppers and OFA Admins, and Data Preppers should be limited to only their own STT